Harvey-OS/harvey
# Reintroduce libhttpd

commit 163b43ebf967ff584c6aacc8d62b03f2ae0a3b8d
Author: Gerard van de Schoot <gvdschoot@gmail.com>
Date:   Fri Mar 4 14:44:52 2016 +0100

    Removed [-Woverflow] warnings

    I do question btw why var `h->bodylen` and `stop` which are type
    uint32_t overflows with ~0UL.

    io.c:30:16: warning: large integer implicitly truncated to unsigned type
    [-Woverflow]
       h->bodylen = ~0UL;

    parse.c:456:10: warning: large integer implicitly truncated to unsigned
    type [-Woverflow]
       stop = ~0UL;

commit 331ffebb5394eb0768cea498e43a2d2c98829e9d
Author: Sevki Hasirci <s@sevki.org>
Date:   Mon Feb 22 00:55:22 2016 +0100

    just the files

commit 48090c023db7a9d0f4f36f3b270dc30080c24ce5
Author: ron minnich <rminnich@gmail.com>
Date:   Sun Feb 14 05:49:03 2016 +0100

    Revert "build: added new build files"
    Does not work at all. Can't build a working system with it yet.

    Let's take this slower, ok? Not break it all next time.

    This reverts commit 4fd75744926150212a0c0cd3c5a2ea51d0637d12.

    Change-Id: I91fce92249dd9d7bf3ca731a124c61a499382dd2

commit 4fd75744926150212a0c0cd3c5a2ea51d0637d12
Author: Sevki <sevki@spotify.com>
Date:   Wed Jan 27 12:46:31 2016 +0100

    build: added new build files

    kernel compiles and crashes

    closes #37

    to test use

            go get -u sevki.org/build/cmd/build
            build //sys/src/9/amd64:harvey
    or

            build -v //sys/src/9/amd64:harvey

    inside a editor like emacs or acme.

    If you are building on OS X you should also have something like .build
    file at the root of the project and add something like

            CC: gcc
            TOOLPREFIX: x86_64-elf-

    Change-Id: Ib6f6156eee1936ceb5f8b0bfa64ed45589195c31
    Signed-off-by: Sevki <sevki@spotify.com>

commit 509545d74c842f94fd908060a879aeaf960c9870
Author: Ronald G. Minnich <rminnich@gmail.com>
Date:   Thu May 5 08:14:37 2016 -0700

    clang: we can now build all of userland and boot

    usbd dies with a suicide, as it seems to try to use floating point
    in a note handler, oops. I bet there is a memcpy and clang generates
    code to use xmm.

    I remove the call to build libz/regression.json because clang got upset
    with it.

    There are many warnings, still, but at least we can build. I am going to get the warnings
    fixed tonight/tomorrow and then add clang to the travis test matrix.

    Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

commit 6533db6114e97d5687bdb6a0f94136c98d009408
Author: Dan Cross <cross@gajendra.net>
Date:   Mon May 1 00:00:16 2017 +0000

    Remove trailing whitespace from source files.

    This is a trivial cleanup (done with a script, Ron!).

    Signed-off-by: Dan Cross <cross@gajendra.net>

commit 9a8098fb117d4b948437686bfb155e11be7ef74c
Author: Gerard van de Schoot <gvdschoot@gmail.com>
Date:   Fri Mar 4 19:04:15 2016 +0100

    Changed const type

commit a9d13806e2923195accce1ddbd0caa6e18ce386b
Author: fuchicar <rafita.fernandez@gmail.com>
Date:   Fri Mar 10 12:33:27 2017 +0100

    Removed ifndef _MPINT from libsec.h

    This is not Harvey style. Removing it implies add #include <mp.h> before each #include <libsec.h>

    Signed-off-by: fuchicar <rafita.fernandez@gmail.com>

commit ba14fa27776ef2371a42607ec04bf50287fbd0be
Author: Ronald G. Minnich <rminnich@gmail.com>
Date:   Fri Nov 13 08:20:41 2015 -0800

    New json format per Giacomo's ideas.

    Instead of this:
    {
            "Name": "a",
            ...
    }

    Do this:
    {
            "Name": {
                    ...
            }
    }

    This is way better. Now the name is visible at the top level of any json viewer, we can
    have multiple stanzas per file, the name is way more easy to find, ... the advantages are legion.

    This includes a change to build.go but not to preen.

    This now works fine, BUILD all produces a working kernel and userland. But someone else needs to verify it.

    Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

    Change-Id: I9b1cf2597e582895b8f3ab127bc9c1d77de96cbd

commit c4abceeea99422a2801485b960b3960373f7ff57
Author: Dan Cross <cross@gajendra.net>
Date:   Wed Apr 19 14:37:30 2017 +0000

    Clean up warnings in libhttpd

    Signed-off-by: Dan Cross <cross@gajendra.net>

commit f72011b4f91fc0387b56b90dd7e9692a4a684842
Merge: ebf1e752d aaec63389
Author: Sevki <s@sevki.org>
Date:   Mon Mar 7 00:42:21 2016 +0100

    Merge branch 'master' into master
